lib/api/api.js

```diff
@@ -316,7 +332,17 @@ api.prototype.getTransactionsToApprove = function(depth, reference, callback) {
 
     var command = apiCommands.getTransactionsToApprove(depth, reference);
 
-    return this.sendCommand(command, callback)
+    return this.sendCommand(command, function (err, tips) {
+      if (adjustDepth && err && err.message.indexOf(REFERENCE_TRANSACTION_TOO_OLD) > -1 && ++depth <= maxDepth) {
```

Comment:
Can we retry with doubling the depth? This would reduce the network calls. However, will increase the load on node.

Reply:
That's an interesting idea. However I think that requests won't be many, most likely just one extra request, if promotion is non-stop and tx is consistent/attached above max depth. We can optimize more by adjusting the depth with a time heuristic...